app/WebParsing/UtsgJsonParser.hs

```diff
@@ -1,83 +1,95 @@
 module WebParsing.UtsgJsonParser
-     (parseTimetable,
-      parseAllCourses,
-      insertAllMeetings) where
+     (parseTimetable, insertAllMeetings) where
 
-import Config (databasePath, orgApiUrl, timetableApiUrl)
+import Config (databasePath, timetableApiUrl)
+import Control.Monad (forM_)
 import Control.Monad.IO.Class (liftIO)
-import Data.Aeson (FromJSON (parseJSON), Object, Value (..), decode, decodeFileStrict, (.!=), (.:?))
-import Data.Aeson.Key (toText)
-import Data.Aeson.KeyMap as KM hiding (insert, map)
-import qualified Data.HashMap.Strict as HM
-import Data.Maybe (catMaybes)
+import Data.Aeson (FromJSON (parseJSON), withObject, Value (..), encode, decode, object, (.=), (.!=), (.:?), (.:))
+import qualified Data.Set as Set
 import qualified Data.Text as T
-import Database.Persist.Sqlite (Filter, SqlPersistM, Update, entityKey, deleteWhere, upsert,
-                                insertMany_, runSqlite, selectFirst, (==.), (=.))
-import Database.Tables (Courses (..), EntityField (..), MeetTime (..), Meeting (..),
-                        Times (..), buildTimes)
-import Network.HTTP.Conduit (simpleHttp)
-
-coursesJson :: FilePath
-coursesJson = "courses.json"
-
-parseAllCourses :: IO ()
-parseAllCourses = do
-    runSqlite databasePath parseCourses
-
-parseCourses :: SqlPersistM ()
-parseCourses = do
-    deleteWhere ([] :: [Filter Times])
-    deleteWhere ([] :: [Filter Meeting])
-    liftIO . print $ T.pack "parsing JSON data"
-    resp <- liftIO $ decodeFileStrict coursesJson
-    let coursesLst :: Maybe (HM.HashMap T.Text (Maybe DB)) = resp
-        courseData = maybe [] (map dbData . catMaybes . HM.elems) coursesLst
-        (_, sections) = unzip courseData
-        meetings = concat sections
-    mapM_ insertMeeting meetings
+import Database.Persist.Sqlite (SqlPersistM, Update, Entity, entityKey, entityVal, deleteWhere, upsert,
+                                insert, insertMany_, runSqlite, selectFirst, selectList, (==.), (=.))
+import Database.Tables (Courses (..), EntityField (..), MeetTime (..), Meeting (..), buildTimes)
+import Network.HTTP.Conduit
```

Comment:
it looks like you removed the explicit import (`simpleHttp`), but always use explicit imports